src/Core/src/Handlers/DatePicker/DatePickerHandler.Android.cs

```diff
@@ -17,15 +17,10 @@ protected override MauiDatePicker CreatePlatformView()
 				HidePicker = HidePickerDialog
 			};
 
-			var date = VirtualView?.Date;
-
-			if (date != null)
-				_dialog = CreateDatePickerDialog(date.Value.Year, date.Value.Month, date.Value.Day);
-
 			return mauiDatePicker;
 		}
 
-		internal DatePickerDialog? DatePickerDialog { get { return _dialog; } }
+		internal DatePickerDialog? DatePickerDialog { get { return _dialog ??= CreateDefaultDatePickerDialog(); } }
```

Comment:
Can we just create the dialog in `ShowPickerDialog` like the time picker? I also see that in time picker, we always re-create the dialog:

```cs
void ShowPickerDialog(int hour, int minute)
{
	_dialog = CreateTimePickerDialog(hour, minute);
	_dialog.Show();
}
```

I see we do set some values in the mappers for the min/max dates, but I feel like that was wrong as we can totally set this in the `setDateLater` part of the `ShowPickerDialog` just like we set the actual date.

I think this whole setup should rather be a lazy loading thing and we set the min/max when the picker is about to appear.